packages/shared-ux/chrome/serverless_projects_documentation.md

```diff
@@ -0,0 +1,289 @@
+
+#### Example
+
+Let's start by seeing an example and we'll detail the different properties below.
+
+```ts
+import { type NavigationTreeDefinition, getPresets } from '@kbn/shared-ux-chrome-navigation';
+
+const navigationTree: NavigationTreeDefinition = {
+  body: [
+    { type: 'recentlyAccessed' }, // Add the recently accessed items
+    {
+      type: 'navGroup', // A top level group
+      id: 'search_project_nav',
+      title: 'Elasticsearch',
+      icon: 'logoElasticsearch',
+      defaultIsCollapsed: false,
+      breadcrumbStatus: 'hidden',
+      children: [
+        {
+          id: 'search_getting_started',
+          title: 'Getting started',
+          link: 'serverlessElasticsearch', // All **internal** links must be deepLinks
+        },
+        {
+          id: 'explore',
+          title: 'Explore', // A nested group with its children
+          children: [
+            {
+              link: 'discover',
+            },
+            {
+              link: 'dashboards',
+            },
+            {
+              link: 'visualize',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      type: 'navGroup',
+      ...getPresets('ml'), // Insert all the machine learning links
+    },
+  ],
+  footer: [
+    {
+      type: 'navGroup',
+      id: 'project_settings_project_nav',
+      title: 'Project settings',
+      icon: 'gear',
+      breadcrumbStatus: 'hidden',
+      children: [
+        {
+          id: 'settings',
+          children: [
+            {
+              link: 'management',
+              title: 'Management',
+            },
+            {
+              id: 'cloudLinkUserAndRoles',
+              cloudLink: 'userAndRoles', // Add an external link to Cloud
+            },
+            {
+              id: 'cloudLinkPerformance',
+              cloudLink: 'performance',
+            },
+          ],
+        },
+      ],
+    },
+  ],
+};
+```
+
+Once the navigation tree is defined we need to
+
+1. Pass it to the `<DefaultNavigation />` component
+2. Set your navigation component in the `serverless` plugin
+
+```tsx
+import { DefaultNavigation, NavigationKibanaProvider } from '@kbn/shared-ux-chrome-navigation';
+
+const createServerlessSearchSideNavComponent =
+  (
+    core: CoreStart,
+    { serverless, cloud }: { serverless: ServerlessPluginStart; cloud: CloudStart }
+  ) =>
+  () => {
+    return (
+      <NavigationKibanaProvider core={core} serverless={serverless} cloud={cloud}>
+        <DefaultNavigation navigationTree={navigationTree} dataTestSubj="svlSearchSideNav" />
+      </NavigationKibanaProvider>
+    );
+  };
+
+// plugin.ts (public) "start()"
+serverless.setSideNavComponent(createComponent(core, { serverless, cloud }));
+```
+
+#### Navigation tree API
```

Comment:
I think we shouldn't focus on this, because we won't be able to maintain this manually. 
I think we should just point to the source code for API definition, make sure we have all these comments there. Also discuss with the team about how we should approach this and what is the current state. We already had some system in place that extracted api definitions to the docs (Here are some example of what we have automatically generated https://docs.elastic.dev/kibana-dev-docs/api/kibanaReact)


Instead we should focus on concepts and examples

Reply:
Since Seb already had this nicely composed manually, I think we should keep it. I added a warning that this might be incomplete and plan to bring up to the discussion about how we should approach api reference in general (automating it)